### PR TITLE
DBAAS-368 MongoDB Database instance Provision failed: fix core dump when no atlas project CR is found

### PR DESCRIPTION
**Pull request by jianrongzhang89**
This PR fixes an issue in Atlas Operator that can cause the operator pod to terminate and restart when:
1) The user creates an inventory with API key that does not have organization owner permission.
2)The user then uses the inventory to create a DBaaSInstance.

Tested successfully in RHPDS dev env.

### CODE REVIEW

File: pkg/controller/atlasinstance/atlasinstance_controller.go [hunk 1]
```diff
@@ -303,6 +303,10 @@ func (r *MongoDBAtlasInstanceReconciler) reconcileAtlasProject(cx context.Contex
 		}
 		// Fetch the project name from newly created CR
 		atlasProject, err = r.getAtlasProject(cx, inst)
+
+		if err == nil && atlasProject == nil {
+			err = errors.New("could not find AtlasProject custom resource")
```
Comment (redhatHameed) [inline pkg/controller/atlasinstance/atlasinstance_controller.go:308]: adding return here WDYT?

Comment (jianrongzhang89) [reply pkg/controller/atlasinstance/atlasinstance_controller.go]: This part is refactored.

Comment (redhatHameed) [reply pkg/controller/atlasinstance/atlasinstance_controller.go]: Okay, I see you are using here Naked returned interesting - https://go.dev/tour/basics/7

**Review by redhatHameed — APPROVED** [review]
Lgtm

**Comment by jianrongzhang89** [comment]
Another PR is submitted. This one is no longer needed

**Pull request closed without merging**